test/common/config/registry_test.cc

```diff
@@ -97,6 +103,79 @@ TEST(RegistryTest, WithDeprecatedFactoryPublished) {
                           ->name());
 }
 
+class TestVersionedFactory : public PublishedFactory {
+public:
+  std::string name() override { return "testing.published.versioned"; }
+};
+
+REGISTER_FACTORY(TestVersionedFactory,
+                 PublishedFactory){FACTORY_VERSION(2, 5, 39, {{"build.label", "alpha"}})};
+
+// Test registration of versioned factory
+TEST(RegistryTest, VersionedFactory) {
+  const auto& factories = Envoy::Registry::FactoryCategoryRegistry::registeredFactories();
+
+  // Expect that the category is present.
+  ASSERT_NE(factories.find("testing.published"), factories.end());
```

Comment:
Why `ASSERT` vs `EXPECT`?

Reply:
The result of find is dereferenced below, so the EXPECT would cause SIGSEGV. This expectation is required for the test to proceed.